src/server/game/Spells/SpellEffects.cpp

```diff
@@ -2449,12 +2449,10 @@ void Spell::EffectSummonType(SpellEffIndex effIndex)
 
                         TempSummonType summonType = (duration <= 0) ? TEMPSUMMON_DEAD_DESPAWN : TEMPSUMMON_TIMED_DESPAWN;
 
-                        uint32 currMinionsCount = m_caster->m_Controlled.size();
-                        uint32 totalNumGuardians = numSummons + currMinionsCount;
                         for (uint32 count = 0; count < numSummons; ++count)
                         {
                             Position pos;
-                            if (totalNumGuardians == 1)
+                            if (count == 0)
```

Comment:
Doing so will bug the multiple spawn of guardians again, practically: when you summon more than one guardian they will overlap (because this code is called 1 time for every summon and the count will always be 0)

A better logic is needed here

Reply:
there is specific place for guarding summoning. not this one.
some spells use this code and are broken because of this

Reply:
we can revert back it for now and improve this logic in future